exonum/src/storage/proof_map_index/key.rs

```diff
@@ -76,53 +67,109 @@ impl ::std::ops::Not for ChildKind {
     }
 }
 
+/// A structure that represents paths to the any kinds of `ProofMapIndex` nodes.
+#[derive(Copy, Clone)]
+pub struct DBKey {
+    bytes: [u8; DB_KEY_SIZE],
+    start: u16,
+}
+
 impl DBKey {
-    /// Create a new bit slice from the given binary data.
-    pub fn leaf<K: ProofMapKey>(key: &K) -> DBKey {
+    /// Create a path from the given key.
+    pub fn new<K: ProofMapKey>(key: &K) -> DBKey {
         debug_assert_eq!(key.size(), KEY_SIZE);
 
-        let mut data = [0; KEY_SIZE];
-        key.write(&mut data);
+        let mut data = [0; DB_KEY_SIZE];
+        data[0] = LEAF_KEY_PREFIX;
+        key.write(&mut data[1..KEY_SIZE + 1]);
+        data[KEY_SIZE + 1] = 0;
```

Comment:
`KEY_SIZE + 1` is written as `DB_KEY_SIZE - 1` down the line. They are equivalent, right? Let's use one of them then?